multinet/api/tasks/process/csv.py

```diff
@@ -0,0 +1,71 @@
+import csv
+from io import StringIO
+from typing import Any, BinaryIO, Dict, List
+
+from celery import shared_task
+from celery.utils.log import get_task_logger
+
+from multinet.api.models import Table, Upload
+
+from .utils import ColumnType, complete_upload, fail_upload_with_message, processor_dict
+
+logger = get_task_logger(__name__)
+
+
+def process_row(row: Dict[str, Any], cols: List[ColumnType]) -> Dict:
+    """Process a CSV row."""
+    new_row = dict(row)
+
+    for col in cols:
+        entry = row.get(col.key)
+
+        # If null entry, skip
+        if entry is None:
+            continue
+
+        process_func = processor_dict.get(col.type)
+        if process_func is not None:
```

Comment:
Ooh, I think you could try using an [assignment expression](https://docs.python.org/3/whatsnew/3.8.html#assignment-expressions) here. (If we're using Python 3.8 or above, of course.)

Reply:
I think we're on `3.8`, I'll try it out.